### PR TITLE
backend/feature/: #760 added date filter for rideList API for BPP das…

### PR DESCRIPTION
**Pull request by saurav2401**
…hboard

## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
added date filter for rideList API for BPP dashboard

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Ride.hs [hunk 1]
```diff
@@ -299,6 +301,8 @@ findAllRideItems merchantId limitVal offsetVal mbBookingStatus mbRideShortId mbC
     let bookingStatusVal = mkBookingStatusVal ride
     where_ $
       booking ^. BookingProviderId ==. val (toKey merchantId)
+        &&. whenJust_ mbFrom (\defaultFrom -> ride ^. RideCreatedAt >=. val defaultFrom)
+        &&. whenJust_ mbTo (\defaultTo -> ride ^. RideCreatedAt >=. val defaultTo)
```
Comment (0utkarsh) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Ride.hs:305]: Should be `<=.`

File: Backend/app/provider-platform/static-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs [hunk 2]
```diff
@@ -65,16 +65,18 @@ rideList ::
   Maybe Text ->
   Maybe Text ->
   Maybe Money ->
+  Maybe UTCTime ->
+  Maybe UTCTime ->
   Flow Common.RideListRes
-rideList merchantShortId mbLimit mbOffset mbBookingStatus mbReqShortRideId mbCustomerPhone mbDriverPhone mbFareDiff = do
+rideList merchantShortId mbLimit mbOffset mbBookingStatus mbReqShortRideId mbCustomerPhone mbDriverPhone mbFareDiff from to = do
```
Comment (0utkarsh) [inline Backend/app/provider-platform/static-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs:71]: We can have `mb` as prefix for these too

File: Backend/app/provider-platform/static-offer-driver-app/Main/src/API/Dashboard/Ride.hs [hunk 3]
```diff
@@ -62,9 +62,11 @@ rideList ::
   Maybe Text ->
   Maybe Text ->
   Maybe Money ->
+  Maybe UTCTime ->
+  Maybe UTCTime ->
   FlowHandler Common.RideListRes
-rideList merchantShortId mbLimit mbOffset mbBookingStatus mbShortRideId mbCustomerPhone mbFareDiff =
-  withFlowHandlerAPI . DRide.rideList merchantShortId mbLimit mbOffset mbBookingStatus mbShortRideId mbCustomerPhone mbFareDiff
+rideList merchantShortId mbLimit mbOffset mbBookingStatus mbShortRideId mbCustomerPhone mbFareDiff from to =
```
Comment (0utkarsh) [inline Backend/app/provider-platform/static-offer-driver-app/Main/src/API/Dashboard/Ride.hs:68]: same

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs [hunk 4]
```diff
@@ -67,16 +67,18 @@ rideList ::
   Maybe Text ->
   Maybe Text ->
   Maybe Money ->
+  Maybe UTCTime ->
+  Maybe UTCTime ->
   Flow Common.RideListRes
-rideList merchantShortId mbLimit mbOffset mbBookingStatus mbReqShortRideId mbCustomerPhone mbDriverPhone mbFareDiff = do
+rideList merchantShortId mbLimit mbOffset mbBookingStatus mbReqShortRideId mbCustomerPhone mbDriverPhone mbFareDiff from to = do
```
Comment (0utkarsh) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs:73]: same

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Dashboard/Ride.hs [hunk 5]
```diff
@@ -62,9 +62,11 @@ rideList ::
   Maybe Text ->
   Maybe Text ->
   Maybe Money ->
+  Maybe UTCTime ->
+  Maybe UTCTime ->
   FlowHandler Common.RideListRes
-rideList merchantShortId mbLimit mbOffset mbBookingStatus mbShortRideId mbCustomerPhone mbFareDiff =
-  withFlowHandlerAPI . DRide.rideList merchantShortId mbLimit mbOffset mbBookingStatus mbShortRideId mbCustomerPhone mbFareDiff
+rideList merchantShortId mbLimit mbOffset mbBookingStatus mbShortRideId mbCustomerPhone mbFareDiff from to =
```
Comment (0utkarsh) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Dashboard/Ride.hs:68]: mb

File: Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/StaticOfferDriver/Ride.hs [hunk 6]
```diff
@@ -103,10 +103,12 @@ rideList ::
   Maybe Text ->
   Maybe Text ->
   Maybe Money ->
+  Maybe UTCTime ->
+  Maybe UTCTime ->
   FlowHandler Common.RideListRes
-rideList merchantShortId apiTokenInfo mbLimit mbOffset mbBookingStatus mbShortRideId mbCustomerPhone mbDriverPhone mbFareDiff = withFlowHandlerAPI $ do
+rideList merchantShortId apiTokenInfo mbLimit mbOffset mbBookingStatus mbShortRideId mbCustomerPhone mbDriverPhone mbFareDiff from to = withFlowHandlerAPI $ do
```
Comment (0utkarsh) [inline Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/StaticOfferDriver/Ride.hs:109]: mb

File: Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Ride.hs [hunk 7]
```diff
@@ -103,10 +103,12 @@ rideList ::
   Maybe Text ->
   Maybe Text ->
   Maybe Money ->
+  Maybe UTCTime ->
+  Maybe UTCTime ->
   FlowHandler Common.RideListRes
-rideList merchantShortId apiTokenInfo mbLimit mbOffset mbBookingStatus mbShortRideId mbCustomerPhone mbDriverPhone mbFareDiff = withFlowHandlerAPI $ do
+rideList merchantShortId apiTokenInfo mbLimit mbOffset mbBookingStatus mbShortRideId mbCustomerPhone mbDriverPhone mbFareDiff from to = withFlowHandlerAPI $ do
```
Comment (0utkarsh) [inline Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Ride.hs:109]: mb

**Review by 0utkarsh — CHANGES_REQUESTED** [review]

**Review by hkmangla — APPROVED** [review]

**Pull request merged**